pkg/test/framework/components/apps/apps.go

```diff
@@ -53,6 +53,7 @@ type App interface {
 	Name() string
 	Endpoints() []AppEndpoint
 	EndpointsForProtocol(protocol model.Protocol) []AppEndpoint
+	EndpointForPort(port int) AppEndpoint
```

Comment:
why is this needed? I'm not sure it makes sense ... we're trying to abstract away the ports to make the API work for various environments.

Reply:
AuthN policy can be set per port, so I want to have test to test on specific port(s). I haven't added them yet, but will do soon.

Reply:
So maybe it will be better being on the interface of the kubernetes apps implementation, rather than the top-level api?

Reply:
But the kubApp is private, right? I cannot cast and use it from external, can't I?

Reply:
I'm suggesting making it public.  Generally, keep it private until there's a reason to make it public ... this seems like a valid reason.

Reply:
Ok. Make the change. Please let me know if that is what you have in mind.